pkg/promclient/multi_api.go

```diff
@@ -101,6 +103,34 @@ type MultiAPI struct {
 	requiredCount   int // number "per key" that we require to respond
 }
 
+// Starts a memory monitoring goroutine that oversees the memory usage of the query and cancels memory intensive ones
+// Takes in the context cancel function and outputs the stop channel that needs to closed gracefully
+func (m *MultiAPI) monitorMemory(cancel func()) chan struct{} {
```

Comment:
does this need a stopchan? Couldn't it just also take the context and select with that too?